deltacat/aws/redshift/model/manifest.py

```diff
@@ -5,19 +5,83 @@
 import logging
 from typing import Any, Dict, List, Optional
 from uuid import uuid4
+from enum import Enum
 
 from deltacat import logs
 
 logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
 
 
+class EntryType(str, Enum):
+    """
+    Enum representing all possible content categories of an manifest entry file
+    """
+
+    DATA = "data"
+    POSITIONAL_DELETE = "positional_delete"
+    EQUALITY_DELETE = "equality_delete"
+
+    @classmethod
+    def get_default(cls):
+        return EntryType.DATA
+
+    @classmethod
+    def list(cls):
+        return [c.value for c in EntryType]
+
+
+class EntryFileParams(dict):
+    """
+    Represents parameters relevant to the underlying contents of manifest entry. Contains all parameters required to support DELETEs
+    equality_column_names: List of column names that would be used to determine row equality for equality deletes.  Relevant only to equality deletes
+    url: Full URI for content file. Should be equal to ManifestEntry URI.  Relevant only to positional deletes
+    position: Ordinal position of a deleted row in the target data file identified by uri, starting at 0. Relevant only to positional deletes
+    """
+
+    @staticmethod
+    def of(
+        entry_type: Optional[EntryType] = EntryType.get_default(),
+        equality_column_names: Optional[List[str]] = None,
+        url: Optional[str] = None,
+        position: Optional[int] = None,
+    ):
+        entry_file_params = EntryFileParams()
+        if entry_type is not None:
+            entry_file_params["entry_type"] = entry_type.value
+        if equality_column_names is not None:
+            entry_file_params["equality_column_names"] = equality_column_names
+        if url is not None:
+            entry_file_params["url"] = url
+        if position is not None:
+            entry_file_params["position"] = position
+        return entry_file_params
+
+    @property
```

Comment:
Why does EntryFileParams has `entry_type` as well?

Reply:
Yes - that's now been removed from the `ManifestEntry` constructor